pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -219,6 +223,19 @@ type CommonPrometheusFields struct {
 	// Number of seconds to wait until a scrape request times out.
 	ScrapeTimeout Duration `json:"scrapeTimeout,omitempty"`
 
+	// The protocols to negotiate during a scrape. It tells clients what
+	// protocol are accepted by Prometheus and with what weight (most wanted is first).
+	// Supported values (case sensitive): PrometheusProto, OpenMetricsText0.0.1,
+	// OpenMetricsText1.0.0, PrometheusText0.0.4.
+	//
+	// If unset, Prometheus uses its default value.
+	//
+	// It requires Prometheus >= v2.48.0.
+	//
+	// +listType=atomic
```

Comment:
Values can't be deduplicated.
```suggestion
	// +listType=set
```